### PR TITLE
Fix list index out of range when padding nested empty lists

### PR DESCRIPTION
**Pull request by qqaatw**
# What does this PR do?

This PR fixes `list index out of range` error when nested empty lists are supplied.

```
from transformers import BertTokenizer

tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")
padded = tokenizer.pad({"input_ids": [[], [], []] })
print(padded)
``` 

```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    padded = tokenizer.pad({"input_ids": [[], [], []] })
  File "/src/transformers/tokenization_utils_base.py", line 2730, in pad
    while len(required_input[index]) == 0:
IndexError: list index out of range
```

@LysandreJik

### CODE REVIEW

**Comment by LysandreJik** [comment]
What is the use-case of passing nested empty lists like this?

**Comment by LysandreJik** [comment]
I would appreciate it greatly if you could take a look at this when you're back @SaulLu!

**Review by SaulLu — COMMENTED** [review]
Thank you very much for your contribution @qqaatw ! 

What you propose makes sense to me! I just have a small request: could you add a test (probably in `test_padding` in the `test_tokenization_common.py` file)

File: tests/test_tokenization_common.py [hunk 1]
```diff
@@ -2977,8 +2977,8 @@ def test_padding(self, max_length=50):
                 input_r = tokenizer_r.encode_plus("This is a input 1")
                 input_r = tokenizer_r.pad(input_r)
 
-                input_p = tokenizer_r.encode_plus("This is a input 1")
-                input_p = tokenizer_r.pad(input_p)
+                input_p = tokenizer_p.encode_plus("This is a input 1")
+                input_p = tokenizer_p.pad(input_p)
```
Comment (qqaatw) [inline tests/test_tokenization_common.py:2981]: I think this is an overlooking when copying test cases.

Comment (SaulLu) [reply tests/test_tokenization_common.py:3001]: oh! Great catch! Thanks! :+1:

**Review by SaulLu — APPROVED** [review]
That is perfect! Thanks for the addition! :raised_hands:

**Pull request merged**